### PR TITLE
[fix](parquet) fix dict page offset zero incompatible behaviour

### PR DESCRIPTION
**Pull request by suxiaogang223**
## Proposed changes
Reason: https://issues.apache.org/jira/browse/ARROW-5322
Java readers(parquet-mr) handles "dictionaryPageOffset = 0" to determine if dictionary page exists where as the C readers uses "has_dictionaryPageOffset" (_isset bit in thrift message) to determine the same resulting in incompatible behaviours.
Therefore, we should consider that dicttionary page exists when both `__isset.dictionary_page_offset` is true and `dictionary_page_offset` is greater than 0.

### CODE REVIEW

**Comment by suxiaogang223** [comment]
run buildall

**Comment by suxiaogang223** [comment]
@CodiumAI-Agent /review

**Comment by suxiaogang223** [comment]
run buildall

File: be/src/vec/exec/format/parquet/vparquet_column_chunk_reader.cpp [hunk 1]
```diff
@@ -59,16 +59,17 @@ ColumnChunkReader::ColumnChunkReader(io::BufferedStreamReader* reader,
           _io_ctx(io_ctx) {}
 
 Status ColumnChunkReader::init() {
-    size_t start_offset = _metadata.__isset.dictionary_page_offset
```
Comment (morningman) [inline be/src/vec/exec/format/parquet/vparquet_column_chunk_reader.cpp:62]: Better extract a method for `metadata.__isset.dictionary_page_offset && _metadata.dictionary_page_offset > 0)`.

**Comment by suxiaogang223** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**